File: xcode/scroll-cell-test/ViewController.m
//
//  ViewController.m
//  scroll-cell-test
//
//  Created by devon blandin on 3/14/14.
//
//

#import "ViewController.h"
#import "ScrollCell.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    [self.view setBackgroundColor:[UIColor whiteColor]];
    
    UITableView *tableView = [[UITableView alloc] initWithFrame: self.view.bounds style:UITableViewStylePlain];
    [tableView setDataSource:self];
    
    [self.view addSubview: tableView];
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return 10;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    return [[ScrollCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"IDENTIFIER"];
}

@end

File: rubymotion/app/controllers/view_controller.rb
class ViewController < UIViewController
  def viewDidLoad
    super

    view.backgroundColor = UIColor.whiteColor

    @table_view            = UITableView.alloc.initWithFrame(view.bounds, style: UITableViewStylePlain)
    @table_view.dataSource = self

    view.addSubview(@table_view)
  end

  def tableView(table_view, numberOfRowsInSection: section)
    10
  end

  def tableView(table_view, cellForRowAtIndexPath: index_path)
    ScrollCell.alloc.initWithStyle(UITableViewCellStyleDefault, reuseIdentifier: 'IDENTIFIER')
  end
end
